packages/nimble-components/karma.conf.js

```diff
@@ -24,6 +25,10 @@ const commonChromeFlags = [
     '--force-prefers-reduced-motion'
 ];
 
+const webpackEnvironmentPlugin = new webpack.EnvironmentPlugin({
+    NODE_ENV: 'production'
```

Comment:
Maybe for tests and even storybook (? not as clear) we should run in development mode to get the extra assertions and potentially catch issues? If it does a bunch of logging then maybe not though..

Reply:
I didn't see any logging for the basic functionality we're adding here in the tests, so I'll change this to `development`. I'm not sure how to turn it on for storybook.

Reply:
Storybook webpack plugins are configured in the main.js, a similar pattern may work there: https://github.com/ni/nimble/blob/main/packages/nimble-components/.storybook/main.js#L31

Reply:
Summary of an offline discussion:
- Storybook is already using `development`, so we won't do anything to change that.
- We will use `production` for our unit tests to ensure we are getting test coverage on the production code of TanStack.
- We aren't sure of the level of risk associated with only running tests (and storybook) against one configuration each, so we will start with the settings in the 2 bullet points above and may need to change these settings in the future.